include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -1299,7 +1299,7 @@ class open_addressing_impl {
    */
   [[nodiscard]] constexpr key_type const& extract_key(value_type const& slot) const noexcept
   {
-    if constexpr (this->has_payload) {
+    if constexpr (std::decay_t<decltype(*this)>::has_payload) {
```

Comment:
There is nothing dependant here, I think this could just be `if constexpr(has_payload)`